src/main/java/gcewing/architecture/SawbenchGui.java

```diff
@@ -135,16 +161,18 @@ void drawShapeSelection() {
     }
 
     void drawSelectedShapeTitle() {
-        Shape shape = te.getSelectedShape();
-        if (shape != null) {
+        int pageIndex = te.getSelectedPageIndex();
+        int shapeIndex = te.getSelectedShapeIndex();
```

Comment:
Either move this call inside body of `if` condition or add range check to `getSelectedShapeIndex`. Otherwise players will receive `OutOfBoundsException` every time when page index is out of bounds.

Reply:
Thank you for spotting this. Added a range check including a method for it in `SawbenchTE` to reduce code duplication.